knowledge/data-sources/notion/src/index.ts

```diff
@@ -109,6 +110,7 @@ async function getAllPagesIteratively(
         pages.set(currentPage.id, {
             page: currentPage,
             path: "",
+            hasChildren: false,
```

Comment:
Is this call to `pages.set` needed anymore? Since we call it again after computing `hasChildren` in the do-while loop below, I don't think this is needed here anymore.

Reply:
Oh I didn't do this in the do while because this is more like an override, and also it will also need to handle the case of failure if get children call failed. So it is probably better to handle it beforehand and override it in do while loop.